### PR TITLE
Add docs for finalizers

### PR DESCRIPTION
**Pull request by fabianvf**
**Description of the change:**
Adds docs for finalizer field in watches.yaml

**Motivation for the change:**
Docs are good

### CODE REVIEW

**Comment by shawn-hurley** [comment]
We might be able to make the **why** use finalizers generic, but the ansible operator does have some very specific functionality that needs to be described. I think that this use of finalizers is pretty specific for how we have implemented the use of finalizers for ansible users. I think that using finalizers here is also pretty different than golang. 

 @fabianvf think you could add a generic why and what are finalizers to like /doc/finalizers.md and then add a link to the ansible specific documentation? I don't think you change anything in the one that you already have. 

Thoughts?

**Comment by LiliC** [comment]
> think you could add a generic why and what are finalizers to like /doc/finalizers.md

Yes I think that would be a helpful addition to the docs!

File: doc/ansible/dev/advanced_options.md [hunk 1]
```diff
@@ -29,6 +30,9 @@ Some features can be overridden per resource via an annotation on that CR. The o
   reconcilePeriod: 5s
   manageStatus: False
   watchDependentResources: False
+  finalizer:
```
Comment (AlexNPavel) [inline doc/ansible/dev/advanced_options.md]: This finalizer example is missing the required name field

File: doc/ansible/dev/finalizers.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
+# Handling deletion events
+
+The default behavior of an Ansible Operator is to delete all resources the operator
+created during reconciliation when a managed resource is marked for deletion. This
+behavior is usually sufficient for applications that exist only in Kubernetes, but
+sometimes it is necessary to perform more complex operations (for example, when
+your an action performed against a third party API needs to be undone). These more
```
Comment (jmrodri) [inline doc/ansible/dev/finalizers.md]: * Change "your an action" to "your action"

**Pull request merged**